src/Sentry/SentryMonitorOptions.cs

```diff
@@ -0,0 +1,140 @@
+/// </summary>
+public class SentryMonitorOptions : ISentryJsonSerializable
+{
+    private SentryMonitorScheduleType _type = SentryMonitorScheduleType.None;
+    private string? _crontab;
+    private int? _interval;
+    private MeasurementUnit.Duration? _unit;
+
+    /// <summary>
+    /// Set Interval
+    /// </summary>
+    /// <param name="cronTab"></param>
+    public void Interval(string cronTab)
+    {
+        if (_type is not SentryMonitorScheduleType.None)
+        {
+            throw new ArgumentException("You tried to set the interval twice. The Check-Ins interval is supposed to be set only once.");
+        }
+
+        _type = SentryMonitorScheduleType.Crontab;
+        _crontab = cronTab;
+    }
+
+    /// <summary>
+    /// Set Interval
+    /// </summary>
+    /// <param name="interval"></param>
+    /// <param name="unit"></param>
+    public void Interval(int interval, MeasurementUnit.Duration unit)
+    {
+        if (_type is not SentryMonitorScheduleType.None)
+        {
+            throw new ArgumentException("You tried to set the interval twice. The Check-Ins interval is supposed to be set only once.");
+        }
+
+        _type = SentryMonitorScheduleType.Interval;
+        _interval = interval;
+
+        // TODO: Should we do some check here to clamp the supported units?
+        _unit = unit;
+    }
+
+    /// <summary>
+    /// The allowed margin of minutes after the expected check-in time that the monitor will not be considered missed for.
+    /// </summary>
+    public TimeSpan? CheckInMargin { get; set; }
+
+    /// <summary>
+    /// The allowed duration in minutes that the monitor may be in_progress for before being considered failed due to timeout.
+    /// </summary>
+    public TimeSpan? MaxRuntime { get; set; }
+
+    /// <summary>
+    /// The number of consecutive failed check-ins it takes before an issue is created.
+    /// </summary>
+    public int? FailureIssueThreshold { get; set; }
+
+    /// <summary>
+    /// The number of consecutive OK check-ins it takes before an issue is resolved.
+    /// </summary>
+    public int RecoveryThreshold { get; set; }
+
+    /// <summary>
+    /// A tz database string representing the timezone which the monitor's execution schedule is in (i.e. "America/Los_Angeles").
+    /// </summary>
+    public string? Timezone { get; set; }
+
+    /// <summary>
+    /// An actor identifier string. This looks like 'user:john@example.com team:a-sentry-team'. IDs can also be used but will result in a poor DX.
+    /// </summary>
+    public string? Owner { get; set; }
+
+    internal SentryMonitorOptions() { }
+
+    /// <inheritdoc />
+    public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
+    {
+        Debug.Assert(_type != SentryMonitorScheduleType.None, "The Monitor Options do not contain a valid interval." +
+                                                              "Please update your monitor options by setting the Interval.");
```

Comment:
Shouldn't this be guarded against by the constructors?

Reply:
The constructor cannot guard against that because the object gets created and only then get's the schedule type set by the user.